commands/box/contract/root.go

```diff
@@ -385,22 +412,43 @@ func deploy(cmd *cobra.Command, args []string) {
 		fmt.Println(cmd.Use)
 		return
 	}
+
 	amount, err := strconv.ParseUint(args[1], 10, 64)
 	if err != nil {
 		fmt.Println("invalid amount: ", err)
 		return
 	}
-	data := args[0]
 	var bytecode string
-	if _, err := os.Stat(data); err == nil {
-		bytes, err := ioutil.ReadFile(data)
+	data := args[0]
+	suffix := path.Ext(data)
+	if suffix == ".sol" {
+		fmt.Println("Have you installed the solidity editor for the version corresponding to the contract?y/n")
+		input, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		if string(input) != "y" {
+			fmt.Println("Please install and contract the corresponding version of solidity editor")
+			fmt.Println("安装教程：https://goethereumbook.org/en/smart-contract-compile/")
+			return
+		}
```

Comment:
1. solidity compiler version
2. no need to use ReadPassword
3. use English tips
4. use program to check automatically, it is not necessary for men to check!!!